src/GT.Core/Services/Impl/GTCompanyService.cs

```diff
@@ -64,26 +65,87 @@ public async Task<CompanyDTO> AddAsync(CompanyDTO dto)
 			}
 		}
 
-		public string AddCompanyLogo(byte[] file)
+		public async Task AddCompanyLogo(CompanyLogoDTO companyLogoDTO)
 		{
-			throw new NotImplementedException();
+			//create folder if not exist
+			string path = Path.Combine(Directory.GetCurrentDirectory(), "wwwroot/img/CompanyLogos");
+			if (!Directory.Exists(path))
+				Directory.CreateDirectory(path);
+
+			if (companyLogoDTO.File == null || companyLogoDTO.CompanyId == null)
+			{
+				_logger.LogWarning($"Can not use null arguments in method: {nameof(AddCompanyLogo)}");
+				return;
+			}
+
+			var company = await GetByIdAsync(companyLogoDTO.CompanyId);
+			if (company == null)
+			{
+				_logger.LogWarning($"Could not find a company in method: {nameof(AddCompanyLogo)}");
+				return;
+			}
+
+			// Check if company already have a stored logo
+			if (company.CompanyLogoId != null || company.CompanyLogoId != String.Empty)
+			{
+				AddFileToFolder(companyLogoDTO.File, Path.Combine(path, company.CompanyLogoId));
+			}
+			else
+			{
+				// Create new ID for entity and image
+				string companyLogoId, fileName;
+				companyLogoId = fileName = Guid.NewGuid().ToString();
+
+				// Update DB entity with new LogoID
+				company.CompanyLogoId = companyLogoId;
+				await UpdateAsync(company, company.Name);
+
+				AddFileToFolder(companyLogoDTO.File, Path.Combine(path, fileName));
+			}
 		}
 
-		public Task DeleteAsync(string companyId)
+		public async Task DeleteAsync(string id)
 		{
-			throw new NotImplementedException();
+			try
+			{
+				await _companyRepository.DeleteAsync(id);
+			}
+			catch (Exception e)
+			{
+				_logger.LogError(e.Message);
+			}
 		}
 
 		public void DeleteCompanyLogo(string companyLogoId)
 		{
-			throw new NotImplementedException();
+			if (companyLogoId == null || companyLogoId.Length <= 0)
+			{
+				_logger.LogWarning($"Arguments cannot be null when using the method: { nameof(DeleteCompanyLogo)}.");
+				return;
+			}
+
+			string path = Path.Combine(Directory.GetCurrentDirectory(), "wwwroot/img/CompanyLogos");
+			string fileNameWithPath = Path.Combine(path, companyLogoId);
+
+			using (var stream = new FileStream(fileNameWithPath, FileMode.Create))
+			{
+				try
+				{
+					File.Delete(fileNameWithPath);
+				}
+				catch (Exception e)
+				{
+					_logger.LogError(e.Message);
+					return;
+				}
+			}
 		}
 
 		public async Task<bool> ExistsByNameAsync(string name)
 		{
 			try
 			{
-				return await _companyRepository.GetAll().Where(e => e.Name == name).AnyAsync();
+				return await _companyRepository.GetAll().Where(c => c.Name == name).AnyAsync();
```

Comment:
For consistency we use 'e' for 'entity' when using lambdas. 

Old implementation will be fine:
`return await _companyRepository.GetAll().Where(e => e.Name == name).AnyAsync();`

Reply:
Fixed to use standard format for project